ui/src/updateChecker.ts

```diff
@@ -0,0 +1,197 @@
+  public notifyEnable() {
+    const isEnabled = svelteStore.get(isEnabledStore);
+    if (isEnabled === null) {
+      notification.primary({
+        message: "Want to check for new versions automatically?",
+        actions: [
+          {
+            label: "Go to settings",
+            handler: () => {
+              router.push(path.settings());
+            },
+          },
+          {
+            label: "Dismiss",
+            handler: () => {
+              isEnabledStore.set(false);
+            },
+          },
+        ],
+      });
+    }
+  }
+
+  // Enable background udpate checking.
+  public enable() {
+    isEnabledStore.set(true);
+
+    this.checkNewVersion();
+    if (this.checkInterval === null) {
+      this.checkInterval = window.setInterval(() => {
+        this.checkNewVersion();
+      }, VERSION_CHECK_INTERVAL);
+    }
+  }
+
+  // Disable background udpate checking.
+  public disable() {
+    isEnabledStore.set(false);
+
+    if (this.checkInterval !== null) {
+      clearInterval(this.checkInterval);
+      this.checkInterval = null;
+    }
+  }
+
+  // Returns if background update checking is enabled
+  public isEnabled(): svelteStore.Readable<boolean> {
+    return svelteStore.derived(isEnabledStore, isEnabled => {
+      if (isEnabled === null) {
+        return false;
+      } else {
+        return isEnabled;
+      }
+    });
+  }
+
+  // A store that holds the `LatestVersionInfo` if this fefature has
+  // been enabled and if there is a newer version available.
+  public newVersion(): svelteStore.Readable<LatestVersionInfo | undefined> {
+    return svelteStore.derived(
+      [this.latestVersionInfo, this.currentVersion],
+      ([latestVersionInfo, currentVersion]) => {
+        if (latestVersionInfo && currentVersion) {
+          if (latestVersionInfo.version === currentVersion) {
+            return undefined;
+          } else {
+            return latestVersionInfo;
+          }
+        } else {
+          return undefined;
+        }
+      }
+    );
+  }
+
+  // Fetch information about the latest version. If that version is
+  // newer than the current version and the user has not been notified
+  // since `VERSION_NOTIFY_SILENCE_INTERVAL` we show a notification.
+  private async checkNewVersion() {
+    let latestVersionInfo;
+    try {
+      latestVersionInfo = await fetchLatestVersion();
+    } catch {
+      return;
+    }
+
+    this.latestVersionInfo.set(latestVersionInfo);
+
+    const { version, announcementUrl } = latestVersionInfo;
+
+    const lastNotified = svelteStore.get(lastNotifiedStore);
+    const now = Date.now();
+    if (
+      lastNotified === null ||
+      now >= lastNotified + VERSION_NOTIFY_SILENCE_INTERVAL
+    ) {
+      notification.primary({
+        message: "There is a new version of Upstream available",
+        actions: [
+          {
+            label: `Check out Version ${version}`,
+            handler: () => {
+              window.open(announcementUrl);
```

Comment:
This doesn't work for me on macOS. Have a look at how it's done in the `<BSOD>` component for the [support button](https://github.com/radicle-dev/radicle-upstream/blob/master/ui/Screen/Bsod.svelte#L30).

Reply:
To make this testable I’ve introduced `ipc.openUrl()` in https://github.com/radicle-dev/radicle-upstream/pull/1550/commits/7b869629a3e7338366a8c350d6d73f4841878ce7